test/e2e/pull_mode_cluster_deleted_immediately_test.go

```diff
@@ -0,0 +1,120 @@
+		var clusterContext string
+		var controlPlane string
+		var parentCommand string
+		var namespace string
+		var restConfig *rest.Config
+		var kubeClient kubernetes.Interface
+
+		// prepare for create cluster
+		ginkgo.BeforeEach(func() {
+			clusterName = "member-e2e-" + rand.String(3)
+			homeDir = os.Getenv("HOME")
+			kubeConfigPath = fmt.Sprintf("%s/.kube/%s.config", homeDir, clusterName)
+			clusterContext = fmt.Sprintf("kind-%s", clusterName)
+			controlPlane = fmt.Sprintf("%s-control-plane", clusterName)
+
+			parentCommand = "karmadactl"
+			namespace = "karmada-system"
+
+			var err error
+			restConfig, err = framework.LoadRESTClientConfig(kubeConfigPath, clusterContext)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+			kubeClient, err = kubernetes.NewForConfig(restConfig)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		})
+
+		// create cluster
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Create cluster: %s", clusterName), func() {
+				err := createCluster(clusterName, kubeConfigPath, controlPlane, clusterContext)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+			ginkgo.DeferCleanup(func() {
+				ginkgo.By(fmt.Sprintf("Deleting clusters: %s", clusterName), func() {
+					err := deleteCluster(clusterName, kubeConfigPath)
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					_ = os.Remove(kubeConfigPath)
+				})
+			})
+		})
+
+		// register pull mode cluster
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("registering cluster: %s", clusterName), func() {
+				opts := register.CommandRegisterOption{
+					DryRun:           false,
+					ClusterNamespace: "karmada-cluster",
+					ClusterName:      clusterName,
+					Context:          clusterContext,
+					KubeConfig:       kubeConfigPath,
+				}
+				err := opts.Run(parentCommand)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+
+		ginkgo.It("Test delete unknown status cluster", func() {
+			ginkgo.By("Checking cluster status collection", func() {
+				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+					cluster, err := framework.FetchCluster(karmadaClient, clusterName)
+					g.Expect(err).ShouldNot(gomega.HaveOccurred())
+					if cluster.Status.KubernetesVersion == "" {
+						return false, nil
+					}
+					if len(cluster.Status.APIEnablements) == 0 {
+						return false, nil
+					}
+					if len(cluster.Status.Conditions) == 0 {
+						return false, nil
+					}
+					if cluster.Status.NodeSummary == nil {
+						return false, nil
+					}
+					if cluster.Status.ResourceSummary == nil || len(cluster.Status.ResourceSummary.AllocatableModelings) == 0 {
+						return false, nil
+					}
+					return true, nil
+				}, pollTimeout, pollInterval).Should(gomega.Equal(true))
+			})
+
+			// delete karmada-agent pod
+			ginkgo.By(fmt.Sprintf("Delete agent pod in cluster: %s", clusterName), func() {
+				framework.RemovePod(kubeClient, namespace, register.KarmadaAgentName)
```

Comment:
Should we remove `karmada-agent` Deployment? The Pod was deleted, it can be recreated.